sketches/src/test/java/org/apache/apex/malhar/sketches/QuantilesEstimatorTest.java

```diff
@@ -0,0 +1,204 @@
+
+    quantilesOp.setup(null);
+    quantilesOp.beginWindow(0);
+
+    for (int i = 0; i < 101; i++) {
+      double r = rand.nextGaussian();
+      quantilesOp.data.process(r);
+      randArray.add(r);
+    }
+
+    quantilesOp.endWindow();
+
+    Collections.sort(randArray);
+
+    Assert.assertEquals("Captures all computed quantiles", sink.collectedTuples.size(), 101);
+    Assert.assertTrue("Computes median correctly", randArray.get(50) == sink.collectedTuples.get(100)[2]);
+  }
+
+  @Test
+  public void testCDF()
+  {
+    QuantilesEstimator quantilesOp = new QuantilesEstimator(128, (short)12345);
+
+    CollectorTestSink<Double> sink = new CollectorTestSink<>();
+    TestUtils.setSink(quantilesOp.cdfOutput, sink);
+
+    quantilesOp.setup(null);
+    quantilesOp.beginWindow(0);
+
+    for (int i = 0; i < 1001; i++) {
+      double r = 0.001 * i;
+      quantilesOp.data.process(r);
+    }
+    quantilesOp.endWindow();
+
+    List<Double> cdfValues = sink.collectedTuples;
+    Assert.assertTrue("Highest CDF value is approx 1.0", cdfValues.get(cdfValues.size() - 1) >= 0.99 &&
+        cdfValues.get(cdfValues.size() - 1) <= 1.0);
+    Assert.assertTrue("Lowest CDF value is approx 0.0", cdfValues.get(0) >= 0.0 &&
+        cdfValues.get(0) <= 0.01);
+  }
+
+  @Test
+  public void testPMF()
+  {
+    QuantilesEstimator quantilesOp = new QuantilesEstimator(128, (short)12345);
+    double[] intervals = {0.0, 0.20, 0.40, 0.60, 0.80, 1.0};
+    quantilesOp.setPmfIntervals(intervals);
+
+    CollectorTestSink<double[]> sink = new CollectorTestSink<>();
+    TestUtils.setSink(quantilesOp.pmfOutput, sink);
+
+    quantilesOp.setup(null);
+    quantilesOp.beginWindow(0);
+    for (int i = 0; i < 1000; i++) {
+      quantilesOp.data.process(0.001 * i);
+    }
+    quantilesOp.endWindow();
+
+    double[] finalPmf = sink.collectedTuples.get(sink.collectedTuples.size() - 1);
+    Assert.assertTrue("Probability Mass between 0.0 and 0.2 is approx 0.2", finalPmf[1] >= 0.19 && finalPmf[1] <= 0.21);
+  }
+
+  public static class Application implements StreamingApplication
+  {
+    public void populateDAG(DAG dag, Configuration conf)
+    {
+      NumberSource source = dag.addOperator("source", NumberSource.class);
+      QuantilesEstimator quantilesOp = dag.addOperator("quantilesEstimator", QuantilesEstimator.class);
+      ConsoleOutputOperator cdfConsole = dag.addOperator("cdf output", ConsoleOutputOperator.class);
+      ConsoleOutputOperator quantilesConsole = dag.addOperator("quantiles output", ConsoleOutputOperator.class);
+      cdfConsole.setSilent(true);
+      quantilesConsole.setSilent(true);
+
+      dag.addStream("random number stream", source.output, quantilesOp.data);
+      dag.addStream("cdf", quantilesOp.cdfOutput, cdfConsole.input);
+      dag.addStream("quantiles", quantilesOp.quantilesOutput, quantilesConsole.input);
+    }
+  }
+
+  @Test
+  public void testInDAG() throws IOException, Exception
```

Comment:
Yup, just verifying the operator can be integrated into a DAG as expected (as per @ilganeli 's comment).